src/Shared/VideoCard/VideoCard.jsx

```diff
@@ -1,29 +1,45 @@
-import React, { useState } from 'react';
-import { MdOutlineWatchLater, MdPlaylistAdd} from "react-icons/md";
+import React, { useEffect, useState } from 'react';
+import { MdOutlineWatchLater, MdPlaylistAdd, MdAddCircle, MdRemoveCircle, MdCancel} from "react-icons/md";
 import { FcLike, FcDislike } from "react-icons/fc";
-import { useLikedVideos, usePlaylist } from '../../Context';
+import { useAlert, useHistory, useLikedVideos, usePlaylist, useWatchlist } from '../../Context';
 import { Link } from 'react-router-dom';
 import { PlayListModal } from '../PlayListModal/PlayListModal';
 import { CgPlayListRemove } from "react-icons/cg";
-
+import { actionAddVideoToHistory } from '../../Helpers/Services/actions';
+import { v4 as uuid } from "uuid";
 
 
 export const VideoCard = (props) =>{
 
     const [isHover, setIsHover] = useState(false)
+    const {alertContent , setAlertContent} = useAlert()
     const {video_data, module , selected_playlist} = props;
     const { likedVideos,
         setLikedVideos,
         addToLikedVideos, removeFromLikedVideos, addVideoToPlaylist  } = useLikedVideos();
-        const { is_module , setis_module, selectedVid, setSelectedVid,isModalHidden, 
+    const { is_module , setis_module, selectedVid, setSelectedVid,isModalHidden, 
             setIsModalHidden, removeVideoFromPlaylist} = usePlaylist();
+    const { addVideoToWatchlater, removeVideoFromWatchlater } = useWatchlist();
+    const { addVideoToHistory, removeVideoFromHistory} = useHistory();
+
+    // useEffect(()=> addVideoToHistory(video_data),[])
  
     return(
         <>
-            
             <div className="card-container card-shadow video-card-container">
-                <button className="btn card-btn-dismiss" onClick={()=> {module==="LikedVideosPage"?removeFromLikedVideos(video_data._id) :addToLikedVideos(video_data)}}>
-                    {module==="LikedVideosPage"?<FcDislike  size='1.5rem' className=''/> :<FcLike size='1.5rem' className=''/>}
+                {/* LIKE / DISLIKE /REMOVE FROM HISTORY */}
+                <button className="btn card-btn-dismiss" 
+                    onClick={()=> {module==="LikedVideosPage"?
```

Comment:
suggestion: i see that youve done chaining of the ternary here. it does reduce code readability from my pov.